cdap-api/src/main/java/co/cask/cdap/api/spark/AbstractSpark.java

```diff
@@ -147,24 +134,16 @@ public final void initialize(SparkClientContext context) throws Exception {
   /**
    * Classes derived from {@link AbstractSpark} can override this method to initialize the {@link Spark}.
    * {@link SparkClientContext} will be available in this method using {@link AbstractSpark#getContext}.
-   * Default implementation of this method calls the deprecated {@link AbstractSpark#beforeSubmit} method.
    * @throws Exception if there is any error in initializing the Spark
    */
   @TransactionPolicy(TransactionControl.IMPLICIT)
   protected void initialize() throws Exception {
-    beforeSubmit(context);
```

Comment:
I believe it's a warning to have an empty method. Replace these with the 'Do nothing by default' comment